src/jmcomic/jm_option.py

```
@@ -58,90 +58,85 @@ def enable_client_cache_on_condition(cls,
+        properties = {}
+        if album:
+            properties.update(album.get_properties_dict())
+        if photo:
+            properties.update(photo.get_properties_dict())
+        return rule.format(**properties)
 
-        return solver_ls
+    @classmethod
+    def parse_detail_rule(cls, album, photo, rule: str):
+        detail = album if rule.startswith('A') else photo
+        return str(DetailEntity.get_dirname(detail, rule[1:]))
```

Ruff (0.8.2)

`DetailEntity` may be undefined, or defined from star imports

(F405)